graphite/graphite.go

```diff
@@ -26,12 +26,30 @@ type Series struct {
 
 type DataPoint []json.Number
 
+type Error struct {
+	subject string
+	body    string
+	err     error
+}
+
+func (e Error) Error() string {
+	return fmt.Sprintf("graphite: %s: %s", e.subject, e.err.Error())
+}
+
+func (e Error) Body() string {
+	return e.body
+}
+
+func NewError(subject, body string, err error) *Error {
+	return &Error{subject, body, err}
+}
+
 func (r *Request) CacheKey() string {
 	targets, _ := json.Marshal(r.Targets)
 	return fmt.Sprintf("graphite-%d-%d-%s", r.Start.Unix(), r.End.Unix(), targets)
 }
 
-func (r *Request) Query(host string) (Response, error) {
+func (r *Request) setUrl(host string) {
```

Comment:
i find it cleaner, especially since url is now a field of req. the Query() function had too much stuff in it